cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -114,10 +116,14 @@ void SparsePoseGraph::AddScan(
     submap_states_.back().trajectory = submaps;
     CHECK_EQ(submap_states_.size(), submap_indices_.size());
   }
+  // ... One call before the first of insertion_submaps was finished
   const mapping::Submap* const finished_submap =
       insertion_submaps.front()->finished_probability_grid != nullptr
           ? insertion_submaps.front()
           : nullptr;
+  // by providing finished_submap (!=nullptr) to ComputeConstraintsForScan
```

Comment:
I thought about it again.
Maybe we can remove the conditional assignment to `finished_submap` and instead conditionally add `ComputeConstraintsForOldScans` as a work item based on the same condition...?

So I suggest to just revert this comment for now and open a new PR with those changes. (Exceptyou see a problem with this upfront)